packages/origin-247-certificate/src/evented-certificate/certificate.facade.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable, Inject } from '@nestjs/common';
+import {
+    CERTIFICATE_COMMAND_REPOSITORY,
+    CertificateCommandRepository
+} from './repositories/CertificateCommand/CertificateCommand.repository';
+import {
+    CERTIFICATE_EVENT_REPOSITORY,
+    CertificateEventRepository
+} from './repositories/CertificateEvent/CertificateEvent.repository';
+import { EventBus } from '@nestjs/cqrs';
+
+import {
+    IClaimCommand,
+    IIssueCommand,
+    ITransferCommand,
+    IIssuedCertificate,
+    IIssueCommandParams
+} from '../types';
+
+import {
+    CertificateIssuedEvent,
+    CertificateTransferredEvent,
+    CertificateClaimedEvent
+} from './events/Certificate.events';
+
+type CertificateCommand = IIssueCommand<unknown> | ITransferCommand | IClaimCommand;
+@Injectable()
+export class OffchainCertificateFacade<T = null> {
```

Comment:
So facade should basically not contain business logic. You can extract this logic into service, and expose it through a facade. Otherwise it's not a facade, so either we change the name or extract this into another service (which at this point doesn't make sense, but will make sense alter when we expose more stuff).

Reply:
Changed to service